pkg/server/controllers.go

```diff
@@ -209,13 +211,41 @@ func (s *Server) installClusterController(clientConfig clientcmdapi.Config, serv
 		cluster.Server = hostURL.String()
 	}
 
+	c := s.cfg.ClusterControllerOptions.Complete(*kubeconfig, s.kcpSharedInformerFactory, s.apiextensionsSharedInformerFactory)
+	cluster, apiresource, err := c.New()
+	if err != nil {
+		return err
+	}
+
 	if err := server.AddPostStartHook("install-cluster-controller", func(context genericapiserver.PostStartHookContext) error {
 		if err := s.waitForCRDServer(context.StopCh); err != nil {
 			return err
 		}
 
-		ctx := goContext(context)
-		return s.cfg.ClusterControllerOptions.Complete(*kubeconfig, s.kcpSharedInformerFactory, s.apiextensionsSharedInformerFactory).Start(ctx)
+		// TODO(sttts): remove CRD creation from controller startup
```

Comment:
So the bizzarre thing is we're starting the informer before creating the types. Why exactly is this what we want?

Reply:
We do want the CRDs to exist first. But that's probably a follow-up to this? (It also applies to workspaces)